addons/medical_treatment/functions/fnc_onMedicationUsage.sqf

```diff
@@ -7,26 +7,27 @@
  * 0: The patient <OBJECT>
  * 1: Medication Treatment classname <STRING>
  * 2: Max dosage (0 to ignore) <NUMBER>
+ * 3: Max Dose Devatiation <NUMBER>
  * 3: Incompatable medication <ARRAY<STRING>>
  *
  * Return Value:
  * None
  *
  * Example:
- * [player, "morphine", 4, [["x", 1]]] call ace_medical_treatment_fnc_onMedicationUsage
+ * [player, "morphine", 4, 2, [["x", 1]]] call ace_medical_treatment_fnc_onMedicationUsage
  *
  * Public: No
  */
 
-params ["_target", "_className", "_maxDosage", "_incompatibleMedication"];
-TRACE_4("onMedicationUsage",_target,_className,_maxDosage,_incompatibleMedication);
+params ["_target", "_className", "_maxDose", "_maxDoseDeviation", "_incompatibleMedication"];
+TRACE_5("onMedicationUsage",_target,_className,_maxDose,_maxDoseDeviation,_incompatibleMedication);
 
 private _overdosedMedications = [];
 
 // Check for overdose from current medication
-if (_maxDosage > 0) then {
+if (_maxDose > 0) then {
     private _currentDose = [_target, _className] call EFUNC(medical_status,getMedicationCount);
-    if (_currentDose >= floor (_maxDosage + round(random(2)))) then {
+    if (_currentDose > _maxDose + (floor random _maxDoseDeviation)) then {
```

Comment:
Needs some math to make it behave in the way described in `ACE_Medical_Treatment.hpp`.
```suggestion
    // Because both {floor random 0} and {floor random 1} return 0
    if (_maxDoseDeviation > 0) then {
        _maxDoseDeviation = _maxDoseDeviation + 1;
    };

    if (_currentDose > _maxDose + (floor random _maxDoseDeviation)) then {
```